assignment7/src/controllers/blog.controllers.ts

```diff
@@ -0,0 +1,134 @@
+import { Request, Response } from "express";
+import Blog from "../models/Blog";
+import {
+  aggregationQuery,
+  blogUpdateValidation,
+  blogValidationSchema,
+  paramIdValidationSchema,
+} from "../validations/blog";
+
+const getAllBlogs = async (req: Request, res: Response) => {
+  try {
+    const query = req.query;
+    if (Object.keys(query).length) {
+      await aggregationQuery.validateAsync(query);
+
+      getBlogByQuery(query)
+        .then((data) => res.status(200).json({ data }))
+        .catch((error: any) => res.status(400).json({ message: error }));
+    } else {
+      const data = await Blog.find();
+
+      res.status(200).json({ data });
+    }
+  } catch (err) {
+    res.status(400).json({ message: err });
+  }
+};
+
+const getBlogByQuery = async (queries: any) => {
+  const { mostRecent, mostPopular, mostInteraction } = queries;
+
+  if (mostRecent) {
+    const data = await Blog.aggregate([
+      {
+        $sort: {
+          creationDate: -1,
+        },
+      },
+    ]);
+
+    return data;
+  } else if (mostPopular) {
+    const data = await Blog.aggregate([
+      {
+        $sort: {
+          "userRating.views": -1,
+        },
+      },
+    ]);
+
+    return data;
+  } else if (mostInteraction) {
+    const data = await Blog.aggregate([
+      {
+        $sort: {
+          "meta.likes": -1,
+        },
+      },
+    ]);
+
+    return data;
+  } else {
+    throw new Error("Please provide valid query");
+  }
+};
+
+const postBlog = async (req: Request, res: Response) => {
+  try {
+    const body = req.body;
+    await blogValidationSchema.validateAsync(body);
```

Comment:
it schema is not valid so will it throw error?

Reply:
yes